airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -301,6 +302,7 @@ def __init__(
         base_container_name: str | None = None,
         deferrable: bool = False,
         poll_interval: float = 2,
+        log_pod_on_failure: bool = False,
```

Comment:
IMO, the default value should be True to keep the existing behaviour as is and then it could be disabled by explicitly setting it to `False` by the calling function.